Snabble/UI/Payment/Online/BaseCheckoutViewController.swift

```diff
@@ -301,6 +295,32 @@ public class BaseCheckoutViewController: UIViewController {
     }
 }
 
+extension BaseCheckoutViewController: PostPaymentManagerDelegate {
+    func postPaymentManager(
+        _ poller: PostPaymentManager,
```

Comment:
Warum ist der poller hier dabei? Die Klasse hat doch gar keine Methoden, die von der Implementation aufgerufen werden könnten, oder?

Reply:
Apple gibt fast immer bei Delegate Methoden den Sender mit. Siehe Beispielsweise [`UITableViewDelegate`](https://developer.apple.com/documentation/uikit/uitableviewdelegate#). Mit Hilfe der Variable können Entwickler beispielsweise prüfen ob es die richtige Instanz ist die feuert.

Habe es noch in `manager` umbenannt 6a771d0dca600f3d749af49067f0780dc190cb2e

Reply:
Klar dass man beim Tableview die instanz manchmal braucht, aber immer nur um darauf wieder Methoden aufzurufen. Hast du diesen Test schon jemals gebraucht? Ich noch nie.

Reply:
Habe ich paar mal verwendet, aber habe dort keine harte Meinung. Der `PostPaymentManager` hat eine Public API und dementsprechend können Methoden aufgerufen werden. Zusätzlich habe ich es auch schon paar mal in einer Codebase gesehen das grundsätzlich durch ein `assert` geprüft wird, ob die Instanz der Delegate Methode mit der lokalen Instanz übereinstimmt. Empfand ich als unnötig, aber war für das Entwicklerteam hilfreich 🤷 

Ich würde gerne versuchen das SDK nicht nur nach unseren Bedürfnissen zu entwickeln, sondern bestmöglich an den Guidelines von Swift und Patterns innerhalb von Apple Frameworks zu orientieren.

Reply:
Genau das frage ich mich halt: was ist das public API, was eine Implementation des Delegates aufrufen kann/will? Bis eben war das nur `start()`, was für den delegate uninteressant ist aufzurufen, und jetzt ist noch `stop()` dazugekommen für dessen Verwendung mir auch kein Beispiel einfällt. Ich hab da aber auch keine besonders starke Meinung zu, ausser dass ich gerne Sachen weglasse wenn sie überflüssig sind :)

Reply:
Hatte es vergessen mit `stop()`. Was willst du? Ich würde es so lassen, da es als Interface für mich verständlich ist.

Das PostPayment kann gestartet und gestoppt werden. Ebenso gibt es 3 Delegate Methoden die ein Update, Complete und eine Bedingung in einem Fehlerfall behandeln. Ob jemand dann beispielsweise ein Stop und ein späteres `start` verwenden möchte kann ich nicht beurteilen.

Reply:
Ich find's wie gesagt überflüssig, und die Verständlichkeit nimmt ohne diese Parameter IMO auch nich ab, aber mir soll's egal sein.